### PR TITLE
Added timer_create and related calls

### PR DESCRIPTION
**Pull request by xonatius**
Added:
* timer_create
* timer_getoverrun
* timer_gettime
* timer_settime
* timer_delete

https://www.freebsd.org/cgi/man.cgi?query=timer_create+&apropos=0&sektion=0&manpath=FreeBSD+9.0-RELEASE&arch=default&format=html

https://www.man7.org/linux/man-pages/man2/timer_create.2.html

https://man.netbsd.org/timer_create.2

### CODE REVIEW

**Comment by joshtriplett** [comment]
Closing and reopening to re-run CI.

File: src/unix/bsd/freebsdlike/mod.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ pub type speed_t = ::c_uint;
 pub type nl_item = ::c_int;
 pub type id_t = i64;
 pub type vm_size_t = ::uintptr_t;
+pub type timer_t = *mut ::c_void;
```
Comment (JohnTitor) [inline src/unix/bsd/freebsdlike/mod.rs:16]: Is `timer_t` `c_void` on DragonFlyBSD as well?
I only found this: https://github.com/DragonFlyBSD/DragonFlyBSD/blob/85f00d18bcc1ab1d11df7193b597e70a43e0896d/sys/cpu/x86_64/include/stdint.h#L125

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ type __pthread_spin_t = __cpu_simple_lock_nv_t;
 pub type vm_size_t = ::uintptr_t;
 pub type lwpid_t = ::c_uint;
 pub type shmatt_t = ::c_uint;
+pub type timer_t = *mut ::c_void;
```
Comment (JohnTitor) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:13]: Hm, seems it's defined as `int`, am I missing something?

https://github.com/NetBSD/src/search?q=_BSD_TIMER_T_

File: src/unix/linux_like/mod.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@ pub type tcflag_t = ::c_uint;
 pub type clockid_t = ::c_int;
 pub type key_t = ::c_int;
 pub type id_t = ::c_uint;
+pub type timer_t = *mut ::c_void;
```
Comment (JohnTitor) [inline src/unix/linux_like/mod.rs:7]: Seems Android and Linux define `__kernel_timer_t` as `int`?

**Review by JohnTitor — COMMENTED** [review]
This PR is marked as WIP but no activities for about a month so leaving some reviews. I have a question about `timer_t` but other structs/functions definitions look good to me.

**Comment by JohnTitor** [comment]
Triage: I'm going to close this PR as inactive and some items are declared now, if there're still missing items and you find some time to work on it again, feel free to submit a new PR for it, thanks!

**Pull request closed without merging**